management/models.py

```diff
@@ -46,12 +46,17 @@ def set_permissions(self):
         if self.permissions_level in ["public", "internal"]:
             assign_perm(view, standard_group, self)
             assign_perm(view, anonymous_user, self)
+            remove_perm(view, self.owner, self)
```

Comment:
I'm not sure I get this. Why are you removing the `view` permission from the owner? Surely the owner should be able to view the object, right?

Also, would this work if there's no owner?